services/cd-service/pkg/repository/repository.go

```diff
@@ -1443,6 +1452,7 @@ type State struct {
 	Commit                 *git.Commit
 	BootstrapMode          bool
 	EnvironmentConfigsPath string
+	ReleaseVersionsLimit   uint
```

Comment:
Non-blocking comment:

I understand why this field was added here; it is the only scope to pass parameters from `Repository` down to `Transform`. Still, I think we shouldn't treat a "State" as a catch-all bucket of parameters, but that's for a different PR.

Reply:
Unfortunately, like you mentioned, `State` is the only way to pass the configuration to the transformers. It might not be the best solution here, but I don't see any other way without completely refactoring the transformers' parameters.